src/builders/indexable-link-builder.php

```diff
@@ -178,6 +181,44 @@ protected function gather_links( $content ) {
 	 * @return string[] An array of urls.
 	 */
 	protected function gather_images( $content ) {
+		$images = $this->gather_images_from_content( $content );
+
+		$featured = $this->get_featured_image();
+		if ( $featured ) {
+			$images[] = $featured;
+		}
+
+		return $images;
+	}
+
+	/**
+	 * Gathers the thumbnail from the current post.
+	 *
+	 * @return string|null An url, if a thumbnail exists.
+	 */
+	protected function get_featured_image() {
+		global $post;
+		if ( ! $post ) {
+			return null;
+		}
+
+		$featured_id = $this->image_helper->get_featured_image_id( $post->ID );
+		if ( ! $featured_id ) {
+			return null;
+		}
+
+		return $this->image_helper->get_attachment_image_source( $featured_id );
```

Comment:
`get_attachment_image_source` _shouldn't_ return an empty result if you just got a featured_id from the same helper, but still: better safe than sorry.
```suggestion
		$featured_image_source = $this->image_helper->get_attachment_image_source( $featured_id );
		
		if( $featured_image_source === '' ) {
				return null;
		}

		return $featured_image_source;
```